### PR TITLE
Attempting to initiate flow that's not in the destination's classpath…

### PR DESCRIPTION
**Pull request by shamsasari**
… returns back a session reject

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/services/statemachine/FlowStateMachineImpl.kt [hunk 1]
```diff
@@ -98,7 +98,12 @@ class FlowStateMachineImpl<R>(override val id: StateMachineRunId,
         logger.debug { "Calling flow: $logic" }
         val startTime = System.nanoTime()
         val result = try {
-            logic.call()
+            logic.call().apply {
```
Comment (mikehearn) [inline node/src/main/kotlin/net/corda/node/services/statemachine/FlowStateMachineImpl.kt]: I'm not sure the .apply here is helping readability.

File: node/src/main/kotlin/net/corda/node/services/statemachine/StateMachineManager.kt [hunk 2]
```diff
@@ -371,11 +365,21 @@ class StateMachineManager(val serviceHub: ServiceHubInternal,
...
+        session.fiber.logger.debug { "Initiated by $sender using ${sessionInit.initiatingFlowClass}" }
         session.fiber.logger.trace { "Initiated from $sessionInit on $session" }
         resumeFiber(session.fiber)
     }
 
+    private fun SessionInit.loadInitiatingFlowClass(): Class<out FlowLogic<*>> {
+        return try {
+            Class.forName(initiatingFlowClass).asSubclass(FlowLogic::class.java)
+        } catch (e: ClassNotFoundException) {
+            throw SessionRejectException("Don't know $initiatingFlowClass", "Cannot find class $initiatingFlowClass")
```
Comment (mikehearn) [inline node/src/main/kotlin/net/corda/node/services/statemachine/StateMachineManager.kt]: What's up with the two different messages?

Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/statemachine/StateMachineManager.kt]: Trying to have a reject message sent to the counter party, and a different log message (for cases we don't want to reveal too much). But here it doesn't make sense.

File: node/src/main/kotlin/net/corda/node/services/statemachine/FlowStateMachineImpl.kt (hunk 1 above)
Comment (shamsasari) [inline node/src/main/kotlin/net/corda/node/services/statemachine/FlowStateMachineImpl.kt]: Yeah.

**Pull request merged**